deploy-service/common/src/main/java/com/pinterest/deployservice/db/DBHostAgentDAOImpl.java

```diff
@@ -63,10 +63,12 @@ public void insert(HostAgentBean hostAgentBean) throws Exception {
     }
 
     @Override
-    public void update(String id, HostAgentBean bean) throws Exception {
-        SetClause setClause = bean.genSetClause();
+    public void updateChanged(String hostId, HostAgentBean oldBean, HostAgentBean newBean)
+            throws SQLException {
+        SetClause setClause =
+                oldBean == null ? newBean.genSetClause() : newBean.genChangedSetClause(oldBean);
         String clause = String.format(UPDATE_HOST_BY_ID, setClause.getClause());
-        setClause.addValue(id);
+        setClause.addValue(hostId);
```

Comment:
passing the `hostId` as a parameter while it's already contained in both beans is a bit weird, but I guess it can serve as outlining how special that field is ? 
I can't tell if I like it or hate it...